optimizer-hints.md

```diff
@@ -1103,3 +1104,31 @@ set tidb_opt_enable_hash_join=off;
 EXPLAIN SELECT /*+ NO_MERGE_JOIN(t1) */ * FROM t1, t2 WHERE t1.a=t2.a;
 ERROR 1815 (HY000): Internal : Can't find a proper physical plan for this query
 ```
+
+### `SET_VAR` does not work in subqueries
+
+`SET_VAR` is used to set a system variable for the current statement and should not be written in a subquery. If you write it in a subquery, `SET_VAR` might not take effect due to the special handling of subqueries.
+
+In the following example, `SET_VAR` is written in the subquery, so it does not take effect.
+
+```sql
+mysql> SELECT @@MAX_EXECUTION_TIME, a FROM (SELECT /*+ SET_VAR(MAX_EXECUTION_TIME=123) */ 1 as a) t;
++----------------------+---+
+| @@MAX_EXECUTION_TIME | a |
++----------------------+---+
+|                    0 | 1 |
++----------------------+---+
+1 row in set (0.00 sec)
+```
+
+In the following example, SET_VAR is not written in a subquery, so it can take effect.
```

Comment:
```suggestion
In the following example, `SET_VAR` is not written in a subquery, so it can take effect.
```
```suggestion
In the following example, SET_VAR is not written in a subquery, so it can take effect.
```